tests/test_indexes.py

```diff
@@ -0,0 +1,290 @@
+        return count
+
+    def test_build_empty_index(self):
+        """Test build empty index."""
+        expectedDescriptorsCount = 0
+        index = self.indexBuilder.buildIndex()
+        assert expectedDescriptorsCount == self.indexBuilder.bufSize
+        self.assertDynamicIndex(index, expectedDescriptorCount=expectedDescriptorsCount, expectedBufSize=0)
+
+    def test_append_descriptor_to_builder(self):
+        """Test append descriptor to index builder."""
+        expectedDescriptorsCount = 1
+        self.indexBuilder.append(self.faceDescriptor)
+        assert expectedDescriptorsCount == self.indexBuilder.bufSize
+        assert expectedDescriptorsCount == self.getCountOfDescriptorsInStorage(self.indexBuilder)
+
+    def test_append_descriptors_batch_to_builder(self):
+        """Test append descriptors batch to index builder."""
+        expectedDescriptorsCount = 2
+        self.indexBuilder.appendBatch(self.faceDescriptorBatch)
+        assert expectedDescriptorsCount == self.indexBuilder.bufSize
+        assert expectedDescriptorsCount == self.getCountOfDescriptorsInStorage(self.indexBuilder)
+
+    def test_get_descriptor_from_builder(self):
+        """Test get descriptor from internal storage."""
+        expectedDescriptorsCount = 2
+        self.indexBuilder.appendBatch(self.faceDescriptorBatch)
+        assert expectedDescriptorsCount == self.indexBuilder.bufSize
+        for idx in range(expectedDescriptorsCount):
+            with self.subTest(case=f"get descriptor with index: {idx}"):
+                descriptor = self.indexBuilder[idx]
+                assert self.faceDescriptorBatch[idx].rawDescriptor == descriptor.rawDescriptor
+
+    def test_get_descriptor_from_builder_bad_index(self):
+        """Test get descriptor with invalid index from internal storage."""
+        self.indexBuilder.appendBatch(self.faceDescriptorBatch)
+        assert len(self.faceDescriptorBatch) == self.indexBuilder.bufSize
+        nonexistentIndex = 2
+        with self.assertRaises(IndexError) as ex:
+            descriptor = self.indexBuilder[nonexistentIndex]
+        assert str(nonexistentIndex) in ex.exception.args[0]
+
+    def test_get_non_default_descriptor_from_builder(self):
+        """Test get non default descriptor from internal storage."""
+        indexBuilder = self.faceEngine.createFaceIndex(descriptorVersion=56)
+        indexBuilder.append(self.faceDescriptor)
+        assert 1 == indexBuilder.bufSize
+        with pytest.raises(LunaSDKException) as ex:
+            descriptor = indexBuilder[0]
+        self.assertLunaVlError(ex, LunaVLError.InvalidInput.format("Invalid input"))
+
+    def test_remove_descriptor_from_builder(self):
+        """Test remove descriptor from internal storage."""
+        expectedDescriptorsCount = 2
+        self.indexBuilder.appendBatch(self.faceDescriptorBatch)
+        assert expectedDescriptorsCount == self.indexBuilder.bufSize
+        for index, expectedBufSize in ((1, 1), (0, 0)):
```

Comment:
index equals bufSize? why do we need tuple?

Reply:
fixed